.github/workflows/functional-test.yml

```diff
@@ -27,6 +27,11 @@ jobs:
 
     runs-on: macos-14
 
+    env:
+      XCODE_VERSION: 15.3
```

Comment:
maybe add a note that in order to change xcode version one must also check for compatible macOS version in the compat matrix